gp-templates/helper-functions.php

```diff
@@ -287,6 +287,100 @@ function textareas( $entry, $permissions, $index = 0 ) {
+function render_notes( $entry, $permissions ) {
+	list( $can_edit, $can_approve ) = $permissions;
+	$notes                          = GP::$notes->get_by_entry( $entry );
+?>
+	<dl>
+		<dt>
+			<?php echo esc_attr__( 'Notes:', 'glotpress' ); ?>
+		</dt>
+		<dd class="notes">
+			<?php
+			foreach ( $notes as $note ) {
+				render_note( $note, $can_edit );
+			}
+			?>
+		</dd>
+	</dl>
+	<dl>
+	<?php
+
+		if ( GP::$permission->current_user_can(
+			'approve',
+			'translation',
+			$entry->id,
+			array(
+				'translation' => $entry,
+			)
+		) ) {
+		echo '<dt><br>' . esc_attr__( 'New Reviewer note:', 'glotpress' ) . '</dt><br>';
+	?>
+			<dt><textarea autocomplete="off" class="foreign-text" name="note[<?php echo esc_attr( $entry->row_id ); ?>]" id="note_<?php echo esc_attr( $entry->row_id ); ?>"></textarea></dt>
+			<dt><button class="add-note" tabindex="-1" data-nonce="<?php echo esc_attr( wp_create_nonce( 'new-note-' . $entry->id ) ); ?>"><?php esc_attr_e( 'Add note', 'glotpress' ); ?></button></dt>
+	<?php
+		}
+	?>
+	</dl>
+<?php
+}
+
+/**
+ * Render the single note interface
+ *
+ * @param GP_Translation $note          The note object.
+ * @param GP_Glossary    $can_edit      Permission of the user.
+ *
+ * @return int
+ */
+function render_note( $note, $can_edit ) {
+?>
+	<div class="note">
+		<?php gp_link_user( get_userdata( $note->user_id ) ); ?>
+		<?php esc_attr_e( 'Commented', 'glotpress' ); ?>
+		<span class="date">
+			<?php
+				/* translators: How much time before was sent the note */
+				echo esc_html( sprintf( __( '%s ago', 'glotpress' ), human_time_diff( strtotime( $note->date_added ), time() ) ) );
+			?>
+		</span>
+		<button class="note-actions" ><?php esc_attr_e( 'edit', 'glotpress' ); ?></button>
+		<div class="note-body">
+			<?php echo nl2br( make_clickable( esc_html( $note->note ) ) ); ?>
+		</div>
+		<?php if ( $can_edit || get_current_user_id() === $note->user_id ) : ?>
+		<div class="note-body edit-note-body" style="display: none;">
+			<textarea autocomplete="off" class="foreign-text" name="edit-note[<?php echo esc_attr( $note->id ); ?>]" id="edit-note-<?php echo esc_attr( $note->id ); ?>"><?php echo esc_html( $note->note ); ?></textarea>
+			<button class="update-note" tabindex="-1" data-note-id="<?php echo esc_attr( $note->id ); ?>" data-nonce="<?php echo esc_attr( wp_create_nonce( 'edit-note-' . $note->id ) ); ?>">
+				<?php esc_attr_e( 'Update Note', 'glotpress' ); ?>
+			</button>
+			<button class="update-cancel" tabindex="-1">
+				<?php esc_attr_e( 'Cancel', 'glotpress' ); ?>
+			</button>
+			<?php if ( $can_edit ) : ?>
+				<button class="delete-note" tabindex="-1" data-note-id="<?php echo esc_attr( $note->id ); ?>" data-nonce="<?php echo esc_attr( wp_create_nonce( 'delete-note-' . $note->id ) ); ?>">
+					<?php esc_attr_e( 'Delete', 'glotpress' ); ?>
+				</button>
+			<?php endif; ?>
+		</div>
+		<?php endif; ?>
+
+	</div>
+<?php
+	return array(
+		'id'   => $note->id,
+		'note' => $note->note,
+	);
```

Comment:
Is there a reason to return a value here?

Reply:
for unit tests to avoid the check the content of the html